Lib/Protocols/IdSSLOpenSSL.pas

```diff
@@ -2365,6 +2365,17 @@ function LoadOpenSSLLibrary: Boolean;
     if not Result then begin
       Exit;
     end;
+//
+// Zdravko Gabrovski - Check for a global openSSL pointers
+//
+{$IFNDEF STATICLOAD_OPENSSL}
+  if ( GlobalIdCrypto <> IdNilHandle ) and ( GlobalIdSSL <> IdNilHandle ) then begin
+    SSLIsLoaded.Value := True;
+    Result := True;
+    Exit;
+  end;
+{$ENDIF}
+

```

Comment:
If the 2 global handles have been assigned, exiting here will bypass Indy's initialization of OpenSSL's thread-locking callbacks. Are you sure this is intended behavior?